supporting_content/dimensions/commitmentdiscounttype.md

```diff
@@ -0,0 +1,17 @@
+# Column: CommitmentDiscountType
+
+## Documentation
+- Microsoft
+  - Azure:  Understand usage details fields: https://learn.microsoft.com/en-us/azure/cost-management-billing/automate/understand-usage-details-fields
+- GCP
+  - Google Commitment Types:  https://cloud.google.com/docs/cuds#spend_versus_resource_commitments (This is the most similar to our selected implementation)
+- AWS
+  - Amazon: Reservation details - https://docs.aws.amazon.com/cur/latest/userguide/reservation-columns.html
+
+## Discussion Topics
+TBC: Another column that describes the type of CUD in the terms of the CSP, eg. 
```

Comment:
I understand there is a discussion about a new dimension PricingModel, whose values could be maybe OnDemand, SavingsPlan, Reservation, Spot, CUD. Would it match the need?

Moreover, if PricingModel is confirmed, the CommitmentDiscountType could be deducted from the PricingModel:
- If PricingModel is SavingsPlan, then CommitmentDiscountType is SpendBased
- If PricingModel is Reservation or CUD, then CommitmentDiscountType is ResourceBased
- If PricingModel is Spot or OnDemand, then CommitmentDiscountType is NULL
So in that case, would the CommitmentDiscountType be still needed as a dedicated explicit dimension?

Reply:
@rodneyjoyce feel free to chime in but I believe there were discussions to try to limit the pricing model to a 'single responsibility' (e.g. CommitmentDiscount rather than the specific program) and then have the type gives the type and a program name for providing the provider specific column for the name given by provider (e.g. Reserved Instances or Committed Use Discounts).

Reply:
Yes, the problem with the Pricing Model is that we could not agree on the list of values, so we decided to start with these more low-level/specific columns. It's a bit of a chicken and egg situation - once we get these columns signed off we will take a look at Pricing Model with fresh eyes.

WRT the point about inferring the Discount Type from another column (e.g. Pricing Model) - this is a balancing act between convenience and redundancy I guess. Practitioners should be able to solve the most common problems and answer the most common questions with as little effort as possible and without having to consider additional business logic...

The other column hinted at here is capturing the Discount Type Name from each cloud provider which would be specific to their implementation (but will still be a specific Commitment Discount Type.